_includes/architecture.md

```diff
@@ -0,0 +1,21 @@
+
+![Blockstack Architecture](/common/images/architecture.png)
+
+Blockchains require consensus among large numbers of people, so they can be slow. Additionally, a blockchain is not designed to hold a lot of data. This means using a blockchain for every bit of data a user might write and store is expensive. For example, image if an application were storing every tweet in the chain.
+
+Blockstack addresses blockchain performance problems by taking a layered approach. At the base of the system is a blockchain and the Blockstack Naming System(BNS). The blockchain  governs ownership of names (identities)  in the system, names such as domain names, username, and applications names.
+
+Names in Blockstack correspond to routing data in the OSI stack. The routing data is stored in the Atlas Peer Network, the second layer. Every core node that joins the Blockstack Network is able to obtain an entire copy of this routing data. Blockstack uses the routing data to associate names (usernames, domains, application names) with a particular storage location.
+
+The final layer is the Gaia Storage System. Gaia works by hosting a _Gaia hub_ in one or more existing storage systems.
```

Comment:
The pink color for italics is making me think they're links. I wonder if giving italics a slightly colored background highlight would prevent this? Or some other style treatment?

Reply:
I'm not sure here what a "storage system" is, especially in relation to a "storage provider" below.

My understanding is that Gaia hubs basically serve to connect applications with storage provider on behalf of users.

Do you mean to refer to a server host (e.g. DigitalOcean, EC2) as the "storage system" where a hub is installed? I wonder if there's a clearer way to lay out these pieces and how they connect to each other.

Reply:
I played around with the pink color. I think it is something we can look into.  For now, I'll leave that part unchanged.  I'll look at tweaking style after it goes out.

Gaia strictly literally provides a hub service and data storage to a user.  Gaia runs as a service on a compute resource (machine) and Gaia typically writes to disk space (storage) that also belong to the same storage provider.  In theory, you could configure the Gaia hub service on one provider and the storage space on an entirely different provider.  Network resources don't care about the physical location they reference connection. (Oversimplification, this leaves aside deeper questions of performance, latency etc).

Further, the use of containers and domain names for a Gaia service configuration means any entity (person or org) running a Gaia hub can switch a Gaia service and storage from one storage provider to another. Technically it is a matter of moving a container and copying disk space (not the same as migration). Since we can envision a world where people offer Gaia hubs to users, these Gaia hub providers will want the flexibility of moving their service from one compute/storage provider to another depending on pricing, performance, or other considerations generally invisible to users of a Gaia hub.

While there may be users that care which storage provider a Gaia hub is on,
It is confusing to use the term "storage provider" except in the sense it is most commonly used. "Cloud storage provider" or "storage provider" for short is terminology (123,000,000 results) meaning companies that offer storage _and also compute resources_.  Storage provider we should reserve for Azure, Dropbox, etc.

We can use "hub provider" or "hub service" for Gaia.  This is why I was complaining about the use of the words "storage provider" in relation to Gaia interfaces. The phrase is confusing. We should say "storage hub" or "hub provider" or "storage hub URL" or "Gaia storage hub."  The hub should be immaterial to us.  The actual provider could become a point of sale _for the Gaia hub provider_ at some point but those services are likely to become mere commodity even to the person running the hub.

@kantai Just a check here please if I've misunderstood.

Reply:
"data hub" or "data storage URL" actually incredibly more user friendly when you think about it.

Reply:
This is where the terminology draft sheet that I owe you would come in handy 😬

Here are the terms that mainly come to mind here for me:

- "Gaia hub": This is indeed a _service_ used by Blockstack users, and possibly set up by them as well. All hubs map one-to-many to storage providers (per below) and are necessary for users to connect to storage providers (i.e. Gaia doesn't connect users to storage providers directly). Gaia hubs can be hosted with the same company / general platform as storage providers (e.g. AWS with Gaia hub on EC2 and storage provider "on" / "as" S3) but don't have to be necessarily (e.g. Gaia hub on EC2, storage provider on DigitalOcean Spaces). We could call these "data hubs" as well, if we want a synonym that's more explicit. Though calling "Gaia hub URLs" also "data storage URLs" seems to risk confusion with "storage providers" below.
- "Storage provider": This is where the user's data "lives" having been written by users via Gaia hubs above (and made available to apps as reads via Gaia hubs as well). Users do not ever connect directly to storage providers – all interactions go through Gaia hubs, and no URLs should ever be set by users in the Blockstack UI for storage providers per se.

It appears we bungled the distinction between these two a bit in [the last modal shown for the new onboarding UI](https://forum.blockstack.org/t/user-choice-of-gaia-hub-now-available-during-onboarding/7026) in that we say "Appy recommends you store your data with this provider" (aka "storage provider") but then guide the user's attention to field for a _Gaia hub_ URL. 

We also generally spoke about how the user should choose or indicate a storage provider for usage (e.g. Blockstack's official provider), and yet to do so, we solicited Gaia hub URLs. Our language is technically pretty consistent (aside from the above), but this is easily confusing for users regardless.

Specifically to this part of the documentation, we seem to be lacking a good name for where Gaia hubs are hosted. I think you're recommending "hub provider", but since "provider" is a word already used for "storage provider", perhaps we should call these "Gaia hub hosts" or "data hub hosts"? I think saying that we run hubs (or "hub services" aka running instances of hubs, to be more specific) on "storage providers" is going to cause a lot of confusion since the name for data's "home" would be the same as the computing environment for a Gaia hub.

Side note – I do think many users are likely to care about the storage provider location and capabilities in addition to the Gaia hub's location and capabilities. In fact, they might care even more for the underlying storage provider than the Gaia hub, since the storage provider dictates several important qualities possibility (e.g. if my underlying storage provider is my Dropbox account, I know how to access it with a web UI outside of Blockstack and it gets synced to my desktop automatically).

That may be true even though lots of _other_ users don't care at all about their storage providers and perhaps more about the Gaia hubs (e.g. user wants to know they can ring the neck of a Gaia hub host they use at any time, should they need their data "out" of the system. Or they're mainly concerned about Gaia hub service uptime and latency).

Reply:
A service such as Dropbox or My PC is uniquely different than an AWS / Digital Ocean. That type of service is offering storage at the individual consumer or small corporate level. While you could store your data there, **you can't run a hub service on a Dropbox, Apple Cloud, or My PC account**. 

https://www.dropboxforum.com/t5/Dropbox/Can-I-install-a-software-program-to-Dropbox/idi-p/41293

There is no operating system on which to run a hub service.  They do this for security reasons.  You can configure the Gaia hub service's configuration to store data on Dropbox et. all  --- but the Gaia URL points to the service, that service writes the data, and must live on a compute resource; AWS, Azure, or eventually a local machine. Caveat to local machine it would need to be up 24/7 and open to the internet. 

As for interacting with the data store outside of the hub service, why would you do that? It is the equivalent of editing a file by touching the bits on your disk. Possible in the future we might have applications that allow it.  There nearest analogy I could think of is moving my 1Password file around.  Sure, I can do that, and have, but I never view or modify the file's contents as that would no doubt cause 1Password to stop functioning.